PSKoans/Data/Advice/Fun/CatFunExample.Advice.json

```diff
@@ -0,0 +1,22 @@
+{
+    "Title": "Cat Fun Example",
+    "Content": [
+        "PowerShell doesn't have to all be about Server administration, ops, or automation.",
+        "It can also be fun!",
```

Comment:
OK, so with these advice files, the cmdlet actually pretty much auto-formats the majority of things. The only thing we need to manually do is make sure we add spaces to the beginning of a line that needs additional indentation (it will have a small amount of indentation by default, and the cmdlet handles wrapping long lines automatically as well.

So, we'll need to keep all text that's going to be in the same paragraph in a single line in the json. It makes for long lines in the json, but the cmdlet can format it much better. 🙂 

Also, if you need a blank line, just add an empty string for that line `"",` -- if you add a newline character it may mess up subsequent lines or formatting.

If you need a hand formatting this, just let me know and I'll give these two a full pass for you. Otherwise, just give these files a once-over so they're consistent with the existing advice files as much as is reasonable. Looks great otherwise, though! 💖

(I really need to add stuff about the advice snippets to CONTRIBUTING.md, I'm gonna go file an issue on that.)

Reply:
> 
> 
> Couple things to tidy up those advice snippets and then I'll head on to review the rest. This is looking lovely, thank you so much for all your hard work here! 💖 😊

No worries 😄 I've been doing PowerShell for ages, and in the last 2 years I've moved on to DevOps and started learning all sorts of weird and wonderful stuff. I've got 40-odd repos all private on my GitHub, but I wanted something public I could show whilst also contributing to an open-source Project. 

I wacked a link to https://github.com/Sudoblark/Powershell_Intro_Training on the PowerShell reddit page, as I originally asked for their help when I made the 1st edition of the course (way back in 2017 oh no I'm old 😭), to thank them and hopefully get someone else interested in training/DevOps hybrid stuff. Anyway someone linked me to your repo and I instantly thought "here's a thing I can get my teeth sunk in to". So it's my pleasure 😄 

I've also been wanting to make an Electron app for a while, and since the GitHub RESTAPI is open I've started work on a codeacademy-style interface to teach PowerShell using PSKoans as the content. 

As for the advice files, I seem to get an error when using blank strings

## Advice File:
```PowerShell
{
    "Title": "Fun with Cat Facts",
    "Content": [
        "PowerShell doesn't have to all be about Server administration, ops, or automation.",
        "",
        "It can also be fun!",
        "We've provided some fun examples below just to give you a taste of what PowerShell can do",
        "Cats",
        "   The below code snippit will query a RESTAPI on the internet that provides some fun facts about cats. It then uses the SpeechSynthesizer to tell you these facts! Feel free to run it on your machine:",
        "function CatFactsGalore{",
        "   Add-Type -AssemblyName System.Speech",
        "    $SpeechSynth = New-Object System.Speech.Synthesis.SpeechSynthesizer",
        "   $URI = 'http://www.catfact.info/api/v1/facts.json?per_page=200'",     
        "   $Random = Get-Random -Maximum 200",
        "   $CatFact = (ConvertFrom-Json (Invoke-WebRequest -Uri $URI)).Facts[$Random]",
        "   $SpeechSynth.Speak('did you know?')",     
        "   $SpeechSynth.Speak($CatFact.details)",
        "   }",
        "CatFactsGalore"
    ]
}
```

## Result
```PowerShell
PS C:\Users\Administrator\Documents\GitHub\Dev\PSKoans> get-advice -Name "CatFunExample.Advice"

Those who came before offer you a fragment of wisdom.

     Fun with Cat Facts

   PowerShell doesn't have to all be about Server administration, ops, or automation.
Write-ConsoleLine : Cannot validate argument on parameter 'InputString'. The argument is null or empty. Provide an argument 
that is not null or empty, and then try the command again.
At C:\Users\Administrator\Documents\GitHub\Dev\PSKoans\PSKoans\Public\Show-Advice.ps1:20 char:33
+         $AdviceObject.Content | Write-ConsoleLine
+                                 ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:String) [Write-ConsoleLine], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,Write-ConsoleLine
 
   It can also be fun!
   We've provided some fun examples below just to give you a taste of what PowerShell can do
   Cats
      The below code snippit will query a RESTAPI on the internet that provides some fun facts about cats. It then uses the
   SpeechSynthesizer to tell you these facts! Feel free to run it on your machine:
   function CatFactsGalore{
      Add-Type -AssemblyName System.Speech
       $SpeechSynth = New-Object System.Speech.Synthesis.SpeechSynthesizer
      $URI = 'http://www.catfact.info/api/v1/facts.json?per_page=200'
      $Random = Get-Random -Maximum 200
      $CatFact = (ConvertFrom-Json (Invoke-WebRequest -Uri $URI)).Facts[$Random]
      $SpeechSynth.Speak('did you know?')
      $SpeechSynth.Speak($CatFact.details)
      }
   CatFactsGalore

PS C:\Users\Administrator\Documents\GitHub\Dev\PSKoans>
```

## With \n
But then if I do ```\n``` instead of ```""```
```PowerShell
PS C:\Users\Administrator\Documents\GitHub\Dev\PSKoans> get-advice -Name "CatFunExample.Advice"

Those who came before offer you a fragment of wisdom.

     Fun with Cat Facts

   PowerShell doesn't have to all be about Server administration, ops, or automation.

   It can also be fun!
   We've provided some fun examples below just to give you a taste of what PowerShell can do
   Cats
      The below code snippit will query a RESTAPI on the internet that provides some fun facts about cats. It then uses the
   SpeechSynthesizer to tell you these facts! Feel free to run it on your machine:
   function CatFactsGalore{
      Add-Type -AssemblyName System.Speech
       $SpeechSynth = New-Object System.Speech.Synthesis.SpeechSynthesizer
      $URI = 'http://www.catfact.info/api/v1/facts.json?per_page=200'
      $Random = Get-Random -Maximum 200
      $CatFact = (ConvertFrom-Json (Invoke-WebRequest -Uri $URI)).Facts[$Random]
      $SpeechSynth.Speak('did you know?')
      $SpeechSynth.Speak($CatFact.details)
      }
   CatFactsGalore

PS C:\Users\Administrator\Documents\GitHub\Dev\PSKoans>
```

Reply:
> Anyway someone linked me to your repo and I instantly thought "here's a thing I can get my teeth sunk in to". So it's my pleasure 😄

I do believe that was me, hi! 😁 Glad you're having fun with it! 😊 

As for the advice files... good to know, I'll have to make sure that gets a mention in the writeup I'll be doing. 🙂 

Looks good overall, then; if you can just make sure that any indentation for those is consistent, and maybe rename the functions shown to something with an approved verb in it (e.g., `Get-CatFact` or maybe `New-CatFact`) -- or perhaps just remove the `function` wrapper and leave the core code -- then those are all good to go.

Reply:
Aha subtle manipulation >.> But yes I am having fun 😃 

Should be renamed and whatnot in the latest commit.